### ISSUE TITLE
Triggers are not activated when there are multiple condition with delay and rising

### ISSUE THREAD

**Issue by bleach31** [esmini/esmini#202]
When there were multiple conditions with delay and conditionEdge="none" , The trigger will be activated at the timing as described in [the implementation](https://github.com/esmini/esmini/blob/master/release_notes.md#2021-06-14-version-2102)

However. when there were multiple conditions with delay and conditionEdge="rising", The trigger will never be activated.

I'm using esmini v2.18.2.

This is a sample scenario of approaching a stopped target, and stop, and finally re-start again.

[delay-rising.xosc.txt](https://github.com/esmini/esmini/files/7782709/delay-rising.xosc.txt)

Two conditions are created for the restart:

```xml
<ConditionGroup>
   <Condition name="DealyRisingCond1" delay="5" conditionEdge="rising">
      <ByEntityCondition>
         <TriggeringEntities triggeringEntitiesRule="all">
            <EntityRef entityRef="Ego" />
         </TriggeringEntities>
         <EntityCondition>
            <SpeedCondition value="1" rule="lessOrEqual" />
         </EntityCondition>
      </ByEntityCondition>
   </Condition>
   <Condition name="DealyRisingCond2" delay="5" conditionEdge="rising">
      <ByEntityCondition>
         <TriggeringEntities triggeringEntitiesRule="all">
            <EntityRef entityRef="Ego" />
         </TriggeringEntities>
         <EntityCondition>
            <RelativeDistanceCondition value="30.1" freespace="false" entityRef="Target" rule="lessOrEqual" relativeDistanceType="longitudinal" coordinateSystem="entity" />
         </EntityCondition>
      </ByEntityCondition>
   </Condition>
</ConditionGroup>
```

The log shows that the timer has expired (So, I think both conditions are true at 18.200sec ), but Ego car did not restart.

```
16.297: DealyRisingCond1 timer expired at 5.01 seconds
18.200: DealyRisingCond2 timer expired at 5.01 seconds
30.017: Trigger /------------------------------------------------
30.017: StopCond == true, 30.0170 > 30.00 edge: none
30.017: Trigger  ------------------------------------------------/
Closing
```

**Comment by bleach31** [comment]
Sorry, there was a mistake in the description.
It' s already been fixed.

**Comment by eknabevcc** [comment]
Good to hear. I actually suspected it was a typo in the description. I found it interesting anyway since condition edge concept is a bit tricky. So I fill in some comments here for reference, even if you already solved it :)

The image in OpenSCENARIO 1.1 [User Guide section 4.4.2. Condition edges](https://www.asam.net/index.php?eID=dumpFile&t=f&f=4596&token=55bca7d8439f2bae072c4dffd1ee544a6d76b786#_conditions_and_triggers) (scroll down a little bit) illustrates how edges works. Edge rising or falling means that the condition will only be true for the very timestep that it switch between true/false. While the edge=none will return be true as long as the condition stay true.

Even if rising/falling is useful for some rare use cases I think that edge=none is good for most cases. 

Image below illustrates a few cases combining two conditions (varying edge settings):
![image](https://user-images.githubusercontent.com/29146839/147878077-c960dd48-10f4-4573-8126-1fe5ec1f15af.png)

Now looking closer to your scenario. As is it's like case 1 in the image. The conditions spikes (rising edges) happens at different timesteps 16.27 and 17.55 respectively, not coinciding hence no trig. Changing DealyRisingCond2 to edge=none will be like image 2 - no trig. But changing DealyRisingCond1 to edge=none while keeping DealyRisingCond2 edge=rising will be like image 3 and trig will happen since the spike is within the window of first condition=true. Also changing both to none will trig (like in image 4).

In summary: The advice is to use edge=none as default and only the rising/falling when needed. It will give expected result in most cases.

Probably your solution was to change edge type on one or both of the conditions. I'll keep issue open for a few days allowing for comments.

**Comment by bleach31** [comment]
Thanks for the detailed explaining the spec of OSC.

now, I think I understand the esmini's behavior of rising and delay condition.

In rising and delay conditions, 

the evaluation results( looks like a spike, 1 case) are just delayed, and

It does not true after timeout, As in the following explanation:

> When timer has expired the condition will always return true

I would like to ask one additional question.

In the none+delay condition,

Once the timer times out, does it not evaluate the signal at that time, and is it always true?

![image](https://user-images.githubusercontent.com/5919566/147878730-287600d1-3172-41c7-9ce7-d090f4cfed9a.png)

**Comment by eknabevcc** [comment]
I never thought so carefully about it, but it's a very relevant and interesting question. Probably the answer is a mix of your alternatives. I would say the (currently implemented) behavior is as follows: When the condition evaluates to true the timer is started, and at times out the "true" value is used as a spike. 

In the image, case 5 is a rising/falling edge that is just shifted in time and combined with the condition of edge none the trig will happen.

Case 6 is two conditions with none edges, the first condition evaluates to true immediately, and that event is shifted 5 sec and will time out within the true-period of the second condition => trig happen. 

Case 7 however will not trig since the shifted spike falls outside the true time window of second condition.

And finally, case 8 shows that the condition is not re-evaluated after delay time out (one could imagine another timer starting at end of the first one..., but that's not the case). So no trig in that specific case.

![edges2](https://user-images.githubusercontent.com/29146839/147940735-733252f7-8913-4576-90a0-d3bbed4d087c.png)

**Comment by bleach31** [comment]
Thank you very much!

> When the condition evaluates to true the timer is started, and at times out the "true" value is used as a spike.

Oh, This is looks different from [this release note](https://github.com/esmini/esmini/blob/master/release_notes.md#2021-06-14-version-2102)   :flushed:

I understood that the conditions with delay regardless of edge type, is given as a spike.


I close this issue, Thank you again.

I will make some example scenario for confirming this behavior of esmini.

When making sample scenario is finished, I will attach this issue for your just information.
